core/src/main/scala/spinal/core/sim/package.scala

```diff
@@ -569,14 +569,15 @@ package object sim {
     */
   implicit class SimBigIntPimper(x: BigInt) {
     def toBytes(bits: Int = -1, endian: Endianness = LITTLE): Array[Byte] = {
-      val raw = x.toByteArray
-      val byteCount = if (bits < 0) raw.length else (bits + 7) / 8
-      assert(raw.length <= byteCount, "Original BigInt has more bytes then bits specified.")
-      val out = Array.fill[Byte](byteCount) { 0 }
-      for (i <- 0 until byteCount) {
-        out(i) = ((x >> i * 8) & 0xff).toByte
-      }
-      if (endian == BIG) out.reverse else out
+      val requiredLen = x.bitLength + (if(x < 0) 1 else 0)
+      assert(bits < 0 || bits >= requiredLen, "Original BigInt has more bytes then bits specified.")
```

Comment:
The condition bits < 0 would lead the default bits = -1 be asserted as fail always.

Reply:
I don´t think so, it asserts that either `bits < 0` (i.e. default - do it automatically) or that bits is large enough for the data as to not cut off data?